crypto/rand/rand_unix.c

```diff
@@ -104,203 +95,111 @@ int RAND_poll(void)
         duration = 1;
         s$sleep(&duration, &code);
 #  else
-#   ifdef OPENSSL_SYS_VOS_IA32
         /* sleep for 1/65536 of a second (15 us).  */
         duration = 1;
         s$sleep2(&duration, &code);
-#   endif                       /* OPENSSL_SYS_VOS_IA32 */
-#  endif                        /* OPENSSL_SYS_VOS_HPPA */
+#  endif
 
-        /* get wall clock time.  */
+        /* get wall clock time, take 8 bits. */
         clock_gettime(CLOCK_REALTIME, &ts);
-
-        /* take 8 bits */
-        v = (unsigned char)(ts.tv_nsec % 256);
+        v = (unsigned char)(ts.tv_nsec & 0xFF);
         RAND_add(&v, sizeof v, 1);
```

Comment:
Given the discussion about RDTSC's randomness contribution, this call to `clock_gettime` should also be reconsidered.  `clock_gettime` will likely provide fewer random bits than RDTSC because it is _at best_ nanosecond resolution and could be way worse.  I've seen single digit microsecond values returned from `clock_getres`.

Is it worth considering something like this for hosts without the POSIX `clock_gettime` function?
```
#ifdef CLOCK_REALTIME
    clock_gettime(CLOCK_REALTIME, ...)
#else
    gettimeofday()
#endif
```